app/services/moderator/delete_user.rb

```diff
@@ -0,0 +1,60 @@
+module Moderator
+  class DeleteUser < ManageActivityAndRoles
+    attr_reader :user, :admin, :user_params
+
+    def initialize(admin:, user:, user_params:)
+      @user = user
+      @admin = admin
+      @user_params = user_params
+    end
+
+    def self.call_deletion(admin:, user:, user_params:)
+      if user_params[:ghostify] == "true"
+        new(user: user, admin: admin, user_params: user_params).ghostify
+      else
+        new(user: user, admin: admin, user_params: user_params).full_delete
+      end
+    end
+
+    def ghostify
+      @ghost = User.find_by(username: "ghost")
+      reassign_articles
+      reassign_comments
+      delete_non_content_activity_and_user
+    end
+
+    def full_delete
+      delete_comments
+      delete_articles
+      delete_non_content_activity_and_user
+    end
+
+    private
+
+    def delete_non_content_activity_and_user
+      delete_user_activity
+      user.unsubscribe_from_newsletters
+      CacheBuster.new.bust("/#{user.username}")
+      user.delete
+    end
+
+    def reassign_comments
+      return unless user.comments.any?
+
+      user.comments.find_each do |comment|
+        comment.update_columns(user_id: @ghost.id)
+        comment.save
+        comment.path
```

Comment:
`comment.path` doesn't seem to make any effect here :thinking:

Reply:
good point. i just changed it to comment.update which i think covers everything I was intending to do. i had forgotten that the comment isn't getting destroyed so we'll want to update as normal including the callbacks?